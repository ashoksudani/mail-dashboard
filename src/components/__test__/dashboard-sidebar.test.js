import React from 'react';
import DashboardSidebar from '../dashboard-sidebar';

let componentWrapper;
let defaultProps = {
  expandSidebar: false
};
beforeEach(() => {
  componentWrapper = mount(<DashboardSidebar {...defaultProps} />);
});

test('it should render expected elements when expandSidebar is false or true', () => {
  expect(componentWrapper).toMatchSnapshot();

  expect(componentWrapper.find('div[className*="md-thin"]').length).toEqual(1);
  expect(componentWrapper.find('i[className*="th large"]').length).toEqual(1);
  expect(componentWrapper.find('i[className*="mail"]').length).toEqual(1);

  componentWrapper.setProps({ expandSidebar: true });
  expect(componentWrapper.find('div[className*="md-wide"]').length).toEqual(1);
  expect(
    componentWrapper.find('div[className="content active"]').length
  ).toEqual(1);
});
