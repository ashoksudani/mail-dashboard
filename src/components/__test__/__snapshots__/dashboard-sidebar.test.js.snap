// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render expected elements when expandSidebar is false or true 1`] = `
<div
  className="ui push left visible sidebar md-thin dashboard-sidebar"
>
  <div
    className="ui vertical labeled icon accordion ui fluid menu"
  >
    <a
      className="active item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="th large icon"
      />
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="mail icon"
      />
    </a>
  </div>
</div>
`;
